### ISSUE TITLE
Docker-in-docker stops working 

### ISSUE THREAD

**Issue by apoorvmote** [microsoft/vscode-dev-containers#728]
First of all I am really enjoying Docker inside docker. Issue is I get following error 

```bash
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

To solve this issue I rebuild the container and docker works fine. Because outside docker cashes the dev container image so build happens instantly. But inside docker looses its cache. And I have to download again. 

Its not just inside docker cashe that I download again. I use golang VSCode extension that downloads all tools again. I use AWS SAM cli that I have to install again. Unfotunately SAM Cli can exlusively be installed with Homebrew. So this is tricky to install but alternative install may come soon that will solve that problem.

I have Windows 10 Professional, WSL2 with Ubuntu 20.04, 6 Core CPU + 16GB RAM. My RAM gets 90%+ over time. So again I keep task manager open and have to restart then with reboot at full load RAM drops to 60%. 

Basically it gets annoying to reboot and reinstall everything every 30 minutes or so. 

I understand simple solution is to upgrade to 32GB RAM. Which I will do soon. But how does codespaces most expensive plan runs fine with 16GB of RAM. 

Even with reboots I am still enjoying docker-in-docker.

**Comment by 2percentsilk** [comment]
Hi there!

We'd be happy to help but we need a few more details from you to investigate what's going on. Could you provide the output of "temp/dockerd.log" as a start?

Regarding your questions around running out RAM locally but not in a codespace with similar specs:
- In short, 16GB goes a bit farther in a codespace because the container has exclusive use of that memory. In your local case, your RAM has to be split between your Windows OS + WSL2 so now you have two operating systems competing for resources. In addition, you might have Outlook + Edge + any other application running locally that is eating into your RAM. With a codespace, we don't have to worry about any of that and so the container has more to work with. Additionally, we've done a bunch of trimming and optimization for the codespace service to ensure we're running as best as we can with the instances we support. Hope that helps!

**Comment by apoorvmote** [comment]
It will take me some time to get the log. I use docker in docker for backend development. Now I am spending time on frontend development. I don't use docker in frontend. I spent some time today to recreate stopping docker. But it didn't stop. I am always switching back and forth between backend and frontend development. So I will post the logs soon. 

Also I guess the only solution is to upgrade RAM. I am really liking the docker-in-docker so I will do the upgrade.

**Comment by apoorvmote** [comment]
Docker crashed again. If I run `cat /tmp/dockerd.log` then I get following output

```bash
time="2021-01-28T03:22:30.257071800Z" level=info msg="Starting up"
failed to start daemon: pid file found, ensure docker is not running or delete /var/run/docker.pid
```
And if I run `docker version` then I get following output

```bash
Client:
 Version:           19.03.14+azure
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        fd3371eb7df1adeceff5935cf3ade0576a0f48d5
 Built:             Sat Oct 24 07:44:17 2020
 OS/Arch:           linux/amd64
 Experimental:      false
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Comment by joshspicer** [comment]
Hey @apoorvmote! I believe i've fixed your problem with https://github.com/microsoft/vscode-dev-containers/commit/f7a3f6a0a6250c7e971b35e7b60b209a989eb4fb.  Please let me know if this helps.

**Comment by apoorvmote** [comment]
@joshspicer Thanks. I am currently working on frontend. I don't use docker 🐳 in frontend. But I took new settings for test drive. And in my short testing docker didn't crash. I will run extensive tests when I am actually working on backend. But for now I am closing this issue.  Thanks again.
